src/Host/Broker/Impl/RemoteUri/RemoteUriHelper.cs

```diff
@@ -1,78 +1,154 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-
 using System;
 using System.Collections.Generic;
-using System.Text;
-using System.Threading.Tasks;
-using System.Net.Http;
 using System.IO;
-using Newtonsoft.Json;
+using System.Net;
+using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
-using System.Net.Http.Headers;
+using Microsoft.Common.Core;
 
 namespace Microsoft.R.Host.Broker.RemoteUri {
     public class RemoteUriHelper {
         public static async Task HandlerAsync(HttpContext context) {
-            BinaryReader reader = new BinaryReader(context.Request.Body);
-            var url = reader.ReadString();
-            var method = reader.ReadString();
-            Dictionary<string, string> headers = JsonConvert.DeserializeObject<Dictionary<string, string>>(reader.ReadString());
+            var url = context.Request.Headers["x-rtvs-url"];
+
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+            request.Method = context.Request.Method;
+            request.UseDefaultCredentials = true;
```

Comment:
I believe this will use Windows authentication, if one is requested. I'm not sure how this will mesh with headers that came from the client (because they will already contain authentication headers if client decided to authenticate). If it overrides them, it'll break things. If it doesn't, it's probably a no-op and can be removed.